kong/cli/utils/services.lua

```diff
@@ -55,6 +56,50 @@ local function prepare_database(configuration)
   return true
 end
 
+local function prepare_working_dir(configuration)
+  local working_dir = configuration.nginx_working_dir
+
+  -- Check if the folder exists
+  if not IO.file_exists(working_dir) then
+    logger:info("Creating working directory at "..working_dir)
+    local _, exit_code = IO.os_execute("mkdir -p "..working_dir)
+    if exit_code ~= 0 then
+      return false, "Cannot create the working directory at "..working_dir
+    else
+      local _, exit_code = IO.os_execute("chmod -R +wr "..working_dir)
```

Comment:
This is the only thing that I don't really like in this PR. It is kinda overriding the user's umask setting, which should be preserved. Ok for checking the permissions before booting, not so ok to mess up with a user's settings.